src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -197,7 +198,60 @@ export class TestDataProvider implements ITestDataProvider {
     return testResults;
   }
 
-  public static filterTestResultsByParam(
+  public async getTestTypesWithTestCodesAndClassification(
+    testTypes: models.TestType[] = [],
+    testTypeParams: models.TestTypeParams
+  ) {
+    testTypes.forEach(async (testType) => {
```

Comment:
We already have an async block in the main function, do we still need it?

Reply:
Actually, are you sure that this works? I know there are problems with async functions within the forEach block. Example: https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404

Reply:
Requested changes are done on 18783